packages/jest-haste-map/src/worker.js

```diff
@@ -52,11 +52,15 @@ export async function worker(data: WorkerMessage): Promise<WorkerMetadata> {
   // Process a package.json that is returned as a PACKAGE type with its name.
   if (filePath.endsWith(PACKAGE_JSON)) {
     content = fs.readFileSync(filePath, 'utf8');
-    const fileData = JSON.parse(content);
-
-    if (fileData.name) {
-      id = fileData.name;
-      module = [filePath, H.PACKAGE];
+    try {
+      const fileData = JSON.parse(content);
```

Comment:
Should we use https://github.com/sindresorhus/parse-json while we're at it? It has `filename` support